plugins/modules/notification/slack.py

```diff
@@ -379,20 +425,38 @@ def main():
     color = module.params['color']
     attachments = module.params['attachments']
     blocks = module.params['blocks']
+    message_id = module.params['message_id']
 
     color_choices = ['normal', 'good', 'warning', 'danger']
     if color not in color_choices and not is_valid_hex_color(color):
         module.fail_json(msg="Color value specified should be either one of %r "
                              "or any valid hex value with length 3 or 6." % color_choices)
 
+    changed = True
+
+    # if updating an existing message, we can check if there's anything to update
+    if message_id is not None:
+        changed = False
+        msg = get_slack_message(module, domain, channel, message_id)
+        for key in ('icon_url', 'icon_emoji', 'link_names', 'color', 'attachments', 'blocks'):
+            if msg.get(key) != module.params.get(key):
+                changed = True
+                break
+        # if check mode is active, we shouldn't do anything regardless.
+        # if changed=False, we don't need to do anything, so don't do it.
+        if module.check_mode or not changed:
+            module.exit_json(changed=changed)
```

Comment:
I think it makes sense to also return `ts` and `channel` here, like in case of a successful edit.